index.md

```diff
@@ -154,6 +154,72 @@ nav[data-toggle='toc'] {
 }
 ```
 
+## Additional options
+
+By default, the plugin finds the first heading level (`<h1>`, then `<h2>`, etc.) that has more than one element and defaults to 1 (for `<h1>`). The identified level becomes the top heading and the plugin supports a single nesting level associated to it, including the subsequent heading only. E.g., if the identified top level is `<h1>`, the nested level will be `<h2>`, which is the next below it; if the top level is `<h2>`, the nested one will be `<h3>`. No additional level to the nested one and no previous level to the top one will be shown.
+
+The plugin allows options to customize this behavior:
+
+* `headings` (which can be a string): if set to a list of headers (in the form *hN,hM,...* where N, M, ... are levels, e.g., `h1,h2,h3`), all of them are searched instead of the first declared one and of its subsequent heading. Example: `headings: 'h1,h2,h3,h4,h5,h6'` (in this case, all of them are searched and not only `<h2>` and `<h3>` in case `<h2>` is the top level).
+* `class`: (which can be a string): if set to a string, its value is used to customize the CSS class that the plugin sets for each description, concatenating it with the related header level. If not set, the default class heading will be `toc-nav-h`. Example: by default, a top level description (`<h1>`) has class`toc-nav-h1`, a second level description (`<h2>`) has class `toc-nav-h2`, etc. If the option `class` is set to e.g. `my-class`, a top level description (`<h1>`) will have class `my-class-h1`, a second level description (`<h2>`) will have class `my-class-h2`, etc. This feature allows adding CSS attributes to customize the descriptions according its header level.
```

Comment:
> While I like the adding of classes to different levels to make styling easier, the customization of said class seems overkill...

Maybe it is indead an overkill if no one is going to use the same root in other places. You can suggest a code rework or do it yourself if wanted.

> Also, what's the behavior when the top-level heading in the sidebar corresponds to an `<h2>`? My assumption is that users would care more about the level in the table of contents than what h* it corresponds to, so I would make it more clear by using toc-nav-level-1 and toc-nav-level-2 or something like that.

If you have top level h2, h1 in the middle of the document will be always shown when setting `headings: 'h1,h2,h3,h4,h5,h6'`, but not shown if leaving this option out.

> so I would make it more clear by using toc-nav-level-1 and toc-nav-level-2 or something like that.

To make sure I got your point, would you change `toc-nav-h` to `toc-nav-level-h`?